nav2_behavior_tree/include/nav2_behavior_tree/bt_action_server_impl.hpp

```diff
@@ -246,6 +251,28 @@ void BtActionServer<ActionT>::executeCallback()
   }
 }
 
+template<class ActionT>
+void BtActionServer<ActionT>::populateErrorCode(typename std::shared_ptr<typename
```

Comment:
This feels pretty hacky and doesn't exactly scale when we start talking about smoother server and other 3rd party server failure codes. 

Some other options
- Add in a flag for which sever this belongs to in the message definition with an enumeration like we did the error messages in the others. That would be extensible so that more servers could be added by the user in closed source and just need to push their ID next (or more to leave room to grow). 
- We change the message enumerations so that they themselves are separated by 100s (or 1000's why not, we have the room in 16 bit) so its immediately clear when trying to look at the error code match where it came from.  That would be extensible so that more servers could be added by the users in closed source and just need to tick all their error codes up another set of 1,000 (or more for room to grow in the main project). Basically, what you did here but in the message definitions themselves so there are no conflicts. Would we want also to have an enum here for the ranges to help translate? I'm not sure. 
- Bitshift stuff around depending on the type of server so a single 16 bit int can represent it, but can separate the server it came from by location 
- (probably not good) Duplicate all of the error code IDs in the NavToPose/similar messages with a prefix on server name and convert here

Reply:
@ayrton04 thoughts? These are all kind of off the cuff.